linera-views/src/backends/metering.rs

```diff
@@ -143,39 +319,117 @@ where
+            .observe(key.len() as f64);
+        let result = self.store.read_value_bytes(key).await?;
+        match &result {
+            None => self
+                .counter
+                .read_value_none_cases
+                .with_label_values(&[])
+                .inc(),
+            Some(value) => self
+                .counter
+                .read_value_value_size
+                .with_label_values(&[])
+                .observe(value.len() as f64),
+        }
+        Ok(result)
     }
 
     async fn contains_key(&self, key: &[u8]) -> Result<bool, Self::Error> {
-        let _metric = self.counter.contains_key.measure_latency();
+        let _latency = self.counter.contains_key.measure_latency();
+        self.counter
+            .contains_key_key_size
+            .with_label_values(&[])
+            .observe(key.len() as f64);
         self.store.contains_key(key).await
     }
 
     async fn contains_keys(&self, keys: Vec<Vec<u8>>) -> Result<Vec<bool>, Self::Error> {
-        let _metric = self.counter.contains_keys.measure_latency();
+        let _latency = self.counter.contains_keys.measure_latency();
+        self.counter
+            .contains_keys_num_entries
+            .with_label_values(&[])
+            .observe(keys.len() as f64);
+        let key_sizes = keys.iter().map(|k| k.len()).sum::<usize>();
+        self.counter
+            .contains_keys_key_sizes
+            .with_label_values(&[])
+            .observe(key_sizes as f64);
         self.store.contains_keys(keys).await
     }
 
     async fn read_multi_values_bytes(
         &self,
         keys: Vec<Vec<u8>>,
     ) -> Result<Vec<Option<Vec<u8>>>, Self::Error> {
-        let _metric = self.counter.read_multi_values_bytes.measure_latency();
+        let _latency = self.counter.read_multi_values_bytes.measure_latency();
+        self.counter
+            .read_multi_values_num_entries
+            .with_label_values(&[])
+            .observe(keys.len() as f64);
+        let key_sizes = keys.iter().map(|k| k.len()).sum::<usize>();
+        self.counter
+            .read_multi_values_key_sizes
+            .with_label_values(&[])
+            .observe(key_sizes as f64);
         self.store.read_multi_values_bytes(keys).await
     }
 
     async fn find_keys_by_prefix(&self, key_prefix: &[u8]) -> Result<Self::Keys, Self::Error> {
-        let _metric = self.counter.find_keys_by_prefix.measure_latency();
-        self.store.find_keys_by_prefix(key_prefix).await
+        let _latency = self.counter.find_keys_by_prefix.measure_latency();
+        self.counter
+            .find_keys_by_prefix_prefix_size
+            .with_label_values(&[])
+            .observe(key_prefix.len() as f64);
+        let result = self.store.find_keys_by_prefix(key_prefix).await?;
+        let mut num_keys = 0;
+        let mut keys_size = 0;
+        for key in result.iterator() {
+            num_keys += 1;
+            keys_size += key?.len();
+        }
```

Comment:
You might be able to do something like this here:
```suggestion
        let keys_size: usize = result.iterator().map(|key| key?.len()).sum();
        let num_keys = result.iterator().count();
```

Reply:
There was a way to do it nicer
```[rust]
        let (num_keys, keys_size) = result
            .iterator()
            .map(|key| key.map(|k| k.len()))
            .collect::<Result<Vec<usize>, _>>()?
            .into_iter()
            .fold((0, 0), |(count, size), len| (count + 1, size + len));
```